storage/postgres/metadata.go

```diff
@@ -68,40 +70,124 @@
 	return meta, nil
 }
 
-func (s *PGStore) SaveMeta(ty string, id string, m core.Metadata) error {
+func (s *PGStore) SaveMeta(id, timestamp, targetType, targetID, key, value string) error {
 	tx, _ := s.Conn().Begin(context.TODO())
 
-	for key, value := range m {
-		ib := sqlbuilder.NewInsertBuilder()
-		ib.InsertInto(s.table("metadata"))
-		ib.Cols(
-			"meta_target_type",
-			"meta_target_id",
-			"meta_key",
-			"meta_value",
-		)
-		ib.Values(ty, id, key, string(value))
+	ib := sqlbuilder.NewInsertBuilder()
+	ib.InsertInto(s.table("metadata"))
+	ib.Cols(
+		"meta_id",
+		"meta_target_type",
+		"meta_target_id",
+		"meta_key",
+		"meta_value",
+		"timestamp",
+	)
+	ib.Values(id, targetType, targetID, key, string(value), timestamp)
 
-		sqlq, args := ib.BuildWithFlavor(sqlbuilder.PostgreSQL)
+	sqlq, args := ib.BuildWithFlavor(sqlbuilder.PostgreSQL)
 
-		_, err := tx.Exec(
-			context.TODO(),
-			sqlq,
-			args...,
-		)
+	_, err := tx.Exec(
+		context.TODO(),
+		sqlq,
+		args...,
+	)
 
-		if err != nil {
-			tx.Rollback(context.TODO())
+	if err != nil {
+		tx.Rollback(context.TODO())
 
-			return err
-		}
+		return err
 	}
 
-	err := tx.Commit(context.TODO())
+	err = tx.Commit(context.TODO())
 
 	if err != nil {
 		return err
 	}
 
 	return nil
 }
+
+func (s *PGStore) FindMeta(q query.Query) (query.Cursor, error) {
+	q.Limit = int(math.Max(-1, math.Min(float64(q.Limit), 100))) + 1
```

Comment:
This ensures the limit value is between 0 and 101. The reason why, I'm not completely sure - it would be worth at least adding a comment to make this readable at a glance. This logic is borrowed from other `FindX` functions such as FindTransaction. Checking git blame it looks like @altitude may have been the original author - Clem do you have any more info on the reason for this limit so that we could document?